src/main/java/org/fenixedu/ulisboa/integration/sas/tasks/sicabe/IngestSasScholarshipSicabe.java

```diff
@@ -0,0 +1,190 @@
+import org.fenixedu.academic.domain.ExecutionYear;
+import org.fenixedu.academic.domain.GrantOwnerType;
+import org.fenixedu.academic.domain.student.PersonalIngressionData;
+import org.fenixedu.academic.domain.student.Registration;
+import org.fenixedu.academic.domain.student.StatuteType;
+import org.fenixedu.academic.domain.student.StudentStatute;
+import org.fenixedu.academic.domain.student.services.StatuteServices;
+import org.fenixedu.academic.domain.util.email.Message;
+import org.fenixedu.academic.domain.util.email.Recipient;
+import org.fenixedu.academic.domain.util.email.ReplyTo;
+import org.fenixedu.bennu.core.domain.Bennu;
+import org.fenixedu.bennu.core.i18n.BundleUtil;
+import org.fenixedu.bennu.scheduler.CronTask;
+import org.fenixedu.bennu.scheduler.annotation.Task;
+import org.fenixedu.ulisboa.integration.sas.domain.CandidacyState;
+import org.fenixedu.ulisboa.integration.sas.domain.SasScholarshipCandidacy;
+import org.fenixedu.ulisboa.integration.sas.domain.SocialServicesConfiguration;
+import org.fenixedu.ulisboa.integration.sas.service.process.AbstractFillScholarshipService;
+import org.fenixedu.ulisboa.integration.sas.service.sicabe.SicabeExternalService;
+
+import pt.ist.fenixframework.Atomic;
+import pt.ist.fenixframework.FenixFramework;
+
+//Force task to be read only and process each report on its own transaction to avoid errors in a report affecting other reports
+@Task(englishTitle = "Ingest SAS Scholarships from SICABE", readOnly = true)
+public class IngestSasScholarshipSicabe extends CronTask {
+
+    @Override
+    public void runTask() throws Exception {
+
+        try {
+            final int beforeSasCandidacies = Bennu.getInstance().getSasScholarshipCandidaciesSet().size();
+            final long beforeWithStateModified = getNumberOfCandidaciesWithModifiedState();
+
+            final SicabeExternalService sicabe = new SicabeExternalService();
+            final ExecutionYear currentExecutionYear = ExecutionYear.readCurrentExecutionYear();
+            sicabe.removeAllCandidaciesWithoutRegistrationAndLoadAllSasCandidacies(currentExecutionYear);
+
+            sicabe.processAllSasScholarshipCandidacies(currentExecutionYear);
+
+            final int afterSasCandidacies = Bennu.getInstance().getSasScholarshipCandidaciesSet().size();
+            final long afterWithStateModified = getNumberOfCandidaciesWithModifiedState();
+
+            HashSet<Registration> listOfWarningToReport = new HashSet<Registration>();
+            updatePersonalIngressionDataAndUpdateStatuteType(currentExecutionYear, listOfWarningToReport);
+
+            boolean newCandidaciesProcesses =
+                    beforeSasCandidacies != afterSasCandidacies || beforeWithStateModified != afterWithStateModified;
+
+            if (newCandidaciesProcesses || listOfWarningToReport.size() > 0) {
+                sendEmailForUser(
+                        BundleUtil.getString(AbstractFillScholarshipService.SAS_BUNDLE,
```

Comment:
I had to move the SasSpringConfiguration class to ulisboa specifications but to break the dependency between legal pt classes and this one i had to move the bundle [variable](https://github.com/qub-it/fenixedu-legal-pt/blob/ca3486502e51dae2786ba4dd0b30bf49a795ffac/src/main/java/org/fenixedu/bennu/SasSpringConfiguration.java#L7) to AbstractFillScholarshipService class and use it instead